internal/types/compare_order.go

```diff
@@ -253,6 +292,31 @@ func compareTypeOrder(a, b any) CompareResult {
 	}
 }
 
+// compareType compares compareTypeOrder then compares
+// number types. The number types int32, int64 and float64
+// are considered different.
```

Comment:
Ther are not only considered different, there is also quite a specific logic in how we sort them.